### PR TITLE
Support of shunt compensator contingency in security analysis

### PR DESCRIPTION
**Pull request by annetill**
Signed-off-by: Anne Tilloy <anne.tilloy@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/LfContingency.java [hunk 1]
```diff
@@ -99,10 +107,18 @@ public double getActivePowerLoss() {
         }
         buses.forEach(b -> branches.addAll(b.getBranches()));
 
+        Set<Pair<LfShunt, Double>> shunts = new HashSet<>(1);
+        for (Pair<String, Double> shuntAndB : propagatedContingency.getShuntIdsToLose()) {
+            LfShunt shunt = network.getShuntById(shuntAndB.getKey());
+            if (shunt != null) {
+                shunts.add(Pair.of(shunt, shuntAndB.getValue()));
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/network/LfContingency.java:114]: The B value seems to be the global one (bus level) and not the one the targeting shunt?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/LfContingency.java:114]: In this case I think there is something tricky to implement in the LfShuntImpl::updateState to find the right section taking into account triggered IIDM shunt

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/network/LfContingency.java:114]: Indeed, in this PR, it is not obvious but I don't support shunt with voltage control on, so it means that I will always have fixed shunt and a global B. That is why it is working... I am going to add a unit test.

File: src/main/java/com/powsybl/openloadflow/network/LfContingency.java [hunk 2]
```diff
@@ -79,7 +87,7 @@ public double getActivePowerLoss() {
         }
 
         // check if contingency split this network into multiple components
-        if (branches.isEmpty()) {
+        if (branches.isEmpty() && propagatedContingency.getShuntIdsToLose().isEmpty()) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/LfContingency.java:90]: @geofjamg here maybe my check is not so clean... it was designed for branch contingencies only. Could you maybe improve it ?

File: src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java [hunk 3]
```diff
@@ -86,14 +96,19 @@ public PropagatedContingency(Contingency contingency, int index, Set<String> bra
...
+            }
         }
     }
 
     public static List<PropagatedContingency> createListForSensitivityAnalysis(Network network, List<Contingency> contingencies) {
         List<PropagatedContingency> propagatedContingencies = new ArrayList<>();
         for (int index = 0; index < contingencies.size(); index++) {
             Contingency contingency = contingencies.get(index);
-            PropagatedContingency propagatedContingency = PropagatedContingency.create(network, contingency, index);
+            PropagatedContingency propagatedContingency = PropagatedContingency.create(network, contingency, index, null);
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java:111]: @geofjamg do you agree with null here?

File: src/main/java/com/powsybl/openloadflow/network/LfContingency.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/LfContingency.java:90]: Looks good, but probably an issue with other injection contingency

**Review by geofjamg — APPROVED** [review]

**Pull request merged**